src/main/java/org/primefaces/component/datagrid/DataGridRenderer.java

```diff
@@ -171,19 +172,27 @@ protected void encodeGrid(FacesContext context, DataGrid grid) throws IOExceptio
         int numberOfRowsToRender = (itemsToRender + columns - 1) / columns;
         int displayedItemsToRender = rowIndex + itemsToRender;
         String columnClass = DataGrid.COLUMN_CLASS + " " + GridLayoutUtils.getColumnClass(columns);
+        String columnInlineStyle = grid.getRowStyle();
+
+        if (!LangUtils.isValueBlank(grid.getRowStyleClass())) {
+            columnClass += " " + grid.getRowStyleClass();
+        }
```

Comment:
Maybe I am not seeing it but I don't see where you are using the columnClass variable?  I don't see you setting it anywhere?